### ISSUE TITLE
Support SMTP via libcurl

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#2704]
Also support of SSL/TLS, authorization and any other features necessary to use the mail.ru’s smtp gate.

**Comment by rtsisyk** [comment]
I propose net.box-style API:

```
smtp = require('smtp').connect(host, port, { user = 'xx', password = 'xx', tls = true })
-- smtp client automatically re-establish connection
-- on the first iteration - re-connect on every sendmail()
msg_id, err = conn.sendmail(from, to, body)
if msg_id == nil then
   print('failed to send email')
else
   print('msg_id', msg_id)
end
conn:close()
```

https://curl.haxx.se/libcurl/c/smtp-ssl.html

**Comment by Totktonada** [comment]
It works good for me with a mail.ru box. Georgy, thanks!

Just two tiny notes:

* `local smtp = require('smtp.client')` (from the commit message) didn’t work for me, but `local smtp = require('smtp.client').new()` does;
* mail.ru box does not store copies of email sent by smtp, so I added the box itself to `bcc` (it is out of topic of smtp client implementation, yep).

I tried the 816e1c37023f39efa7da2e85e0c786d8730c6bda patch upon Tarantool 1.7.5.101.
